File: apps/server/internal/models/storage_account.go
package models

import (
	"encoding/json"
	"time"

	"github.com/google/uuid"
)

// StorageAccount is one connected provider account inside a workspace — a
// single provider (e.g. Google) may appear several times with different
// accounts. Credentials never live on this struct: they are encrypted into
// storage_account_secrets and only decrypted on demand by the repository.
type StorageAccount struct {
	ID                uuid.UUID       `db:"id" json:"id"`
	OrganizationID    uuid.UUID       `db:"organization_id" json:"organization_id"`
	WorkspaceID       uuid.UUID       `db:"workspace_id" json:"workspace_id"`
	ProviderID        uuid.UUID       `db:"provider_id" json:"provider_id"`
	OwnerUserID       uuid.UUID       `db:"owner_user_id" json:"owner_user_id"`
	DisplayName       string          `db:"display_name" json:"display_name"`
	AccountEmail      *string         `db:"account_email" json:"account_email,omitempty"`
	ExternalAccountID string          `db:"external_account_id" json:"external_account_id"`
	Settings          json.RawMessage `db:"settings" json:"settings,omitempty"`
	Status            string          `db:"status" json:"status"`
	LastSyncedAt      *time.Time      `db:"last_synced_at" json:"last_synced_at,omitempty"`
	DeletedAt         *time.Time      `db:"deleted_at" json:"deleted_at,omitempty"`
	CreatedAt         time.Time       `db:"created_at" json:"created_at"`
	UpdatedAt         time.Time       `db:"updated_at" json:"updated_at"`
}

File: apps/app/lib/api/models/storage-account.ts
import type { ISO8601DateString } from '@/types/time'

export type StorageAccountStatus =
  | 'pending_auth'
  | 'active'
  | 'expired'
  | 'revoked'
  | 'error'
  | string

export interface StorageAccount {
  id: string
  organization_id: string
  workspace_id: string
  provider_id: string
  owner_user_id: string
  display_name: string
  account_email?: string
  external_account_id: string
  settings?: Record<string, unknown>
  status: StorageAccountStatus
  last_synced_at?: ISO8601DateString | null
  deleted_at?: ISO8601DateString | null
  created_at: ISO8601DateString
  updated_at: ISO8601DateString
}
